core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -4111,6 +4111,12 @@ private Sql checkSubQuery(String sql) {
     checkSubQuery(sql).withLateDecorrelation(true).check();
   }
 
+  @Test public void testSomeWithEquality() {
+    final String sql = "select * from emp e1\n"
+        + "  where e1.deptno = SOME (select deptno from dept)";
+    checkSubQuery(sql).withLateDecorrelation(true).check();
+  }
```

Comment:
Where and how do you deal with `=ANY` since your commit message looks like handle it? Do I miss something?

Reply:
@chunweilei `=ANY/SOME` rewrite is done in `RexSubquery.some`. Please look at `core/src/main/java/org/apache/calcite/rex/RexSubQuery.java` changes

Reply:
@chunweilei `=ANY` is rewritten into IN.

Reply:
Got it.